thentos-core/src/Thentos/Action.hs

```diff
@@ -278,18 +262,17 @@ changePassword uid old new = do
     _ <- _lookupUserCheckPassword (T.lookupAnyUser uid) old
     hashedPw <- hashUserPass'P new
     guardWriteMsg "changePassword" (RoleAdmin \/ UserA uid %% RoleAdmin /\ UserA uid)
-    query'P $ T.updateUserField uid (T.UpdateUserFieldPassword hashedPw)
+    query'P $ T.changePassword uid hashedPw
+
 
--- | Unconditionally change the password of a user.
---
 -- SECURITY: As a caller, you have to make sure that the user is properly authenticated and
--- authorized to change the password before calling this function!
+-- authorized to change the password before calling this function
 -- FIXME Reconsider/fix security model once our usage of LIO or a suitable replacement has
--- been clarified.
+-- been clarified
 changePasswordUnconditionally :: UserId -> UserPass -> Action e ()
```

Comment:
At some point in the distant past I tried to introduce a naming convention for actions that should remain local to a module because they do not enforce any satisfactory lio policy.  I'll try that again now.